src/components/AddressSearch/index.js

```diff
@@ -94,10 +94,11 @@ const defaultProps = {
 const AddressSearch = (props) => {
     const [displayListViewBorder, setDisplayListViewBorder] = useState(false);
     const containerRef = useRef();
-    const query = {language: props.preferredLocale, types: 'address'};
-    if (props.isLimitedToUSA) {
-        query.components = 'country:us';
-    }
+
+    // To avoid regeneration of query object on each re-renders
+    const query = useMemo(() => (
+        {language: props.preferredLocale, types: 'address', ...(props.isLimitedToUSA && {components: 'country:us'})}
+    ), [props.preferredLocale, props.isLimitedToUSA]);
```

Comment:
```suggestion
    const query = useMemo(() => ({
        language: props.preferredLocale,
        types: 'address',
        components: props.isLimitedToUSA ? 'country:us' : undefined,
    }), [props.preferredLocale, props.isLimitedToUSA]);
```

Just for visibility and I think the comment is not necessary here.